sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala

```diff
@@ -224,7 +224,8 @@ class SQLQueryTestSuite extends QueryTest with SharedSQLContext {
   private def runTest(testCase: TestCase): Unit = {
     val input = fileToString(new File(testCase.inputFile))
 
-    val (comments, code) = input.split("\n").partition(_.startsWith("--"))
+    val (comments, codeMaybeWithComment) = input.split("\n").partition(_.trim.startsWith("--"))
+    val code = codeMaybeWithComment.map(_.split("--").head)
```

Comment:
Add `split` to fix these comments error:
Comment for ` SELECT date '5874898-01-01'`:
https://github.com/apache/spark/blob/2d74f14d74e7b24109f347822600ebf9819b04c1/sql/core/src/test/resources/sql-tests/inputs/pgSQL/date.sql#L200
But the golden file is:
https://github.com/apache/spark/blob/a5a5da78cfbb1b439c139fd2ffd5744fc60ebafe/sql/core/src/test/resources/sql-tests/results/pgSQL/date.sql.out#L484-L507

Reply:
oh... really weird output... nice catch.